src/Relude/Extra/Enum.hs

```diff
@@ -23,6 +26,27 @@ import Relude
 universe :: (Bounded a, Enum a) => [a]
 universe = [minBound .. maxBound]
 
+{- | Create a function that parses values of type @a@
+
+>>> data Color = Red | Green | Blue deriving (Show, Enum, Bounded)
+>>> parse = inverseMap :: (Color -> String) -> String -> Maybe Color
```

Comment:
This example is not completely precise. To get desired performance level, you need to apply function partially. So `parse` should be implemented as `parse = inverseMap show :: String -> Maybe Color`. And I think that this also should be described in documentation.